packages/utils/messages.ts

```diff
@@ -256,6 +286,97 @@ export const decodeStargateMessage = ({
   },
 })
 
+// Take a CosmWasm msg and return the protobuf encoded version
+export const cosmwasmToProtobuf = (
+  msg: CosmosMsgFor_Empty,
+  address?: string
+): { typeUrl: string; value: Uint8Array } | undefined => {
+  let encoded
+  if ('staking' in msg) {
+    if ('delegate' in msg.staking) {
+      encoded = encodeRawProtobufMsg({
+        typeUrl: '/cosmos.staking.v1beta1.MsgDelegate',
+        value: {
+          amount: msg.staking.delegate.amount,
+          delegatorAddress: address,
+          validatorAddress: msg.staking.delegate.validator,
+        },
+      })
+    } else if ('redelegate' in msg.staking) {
+      encoded = encodeRawProtobufMsg({
+        typeUrl: '/cosmos.staking.v1beta1.MsgBeginRedelegate',
+        value: {
+          delegatorAddress: address,
+          validatorSrcAddress: msg.staking.redelegate.dst_validator,
+          validatorDestAddress: msg.staking.redelegate.src_validator,
+          amount: msg.staking.redelegate.amount,
+        },
+      })
+    } else if ('undelegate' in msg.staking) {
+      encoded = encodeRawProtobufMsg({
+        typeUrl: '/cosmos.staking.v1beta1.MsgBeginRedelegate',
+        value: {
+          delegatorAddress: address,
+          validatorAddress: msg.staking.undelegate.validator,
+          amount: msg.staking.undelegate.amount,
+        },
+      })
+    }
+  } else if ('distribution' in msg) {
+    if ('withdraw_delegator_reward' in msg.distribution) {
+      encoded = encodeRawProtobufMsg({
+        typeUrl: '/cosmos.distribution.v1beta1.MsgWithdrawDelegatorReward',
+        value: {
+          delegatorAddress: address,
+          validatorAddress:
+            msg.distribution.withdraw_delegator_reward.validator,
+        },
+      })
+    }
+  } else if ('bank' in msg) {
+    if ('send' in msg.bank) {
+      encoded = encodeRawProtobufMsg({
+        typeUrl: '/cosmos.bank.v1beta1.MsgSend',
+        value: {
+          fromAddress: address,
+          toAddress: msg.bank.send.to_address,
+          amount: msg.bank.send.amount,
+        },
+      })
+    }
+  } else if ('wasm' in msg) {
+    if ('execute' in msg.wasm) {
+      encoded = encodeRawProtobufMsg({
+        typeUrl: '/cosmwasm.wasm.v1.MsgExecuteContract',
+        value: {
+          sender: address,
+          contract: msg.wasm.execute.contract_addr,
+          funds: msg.wasm.execute.funds,
+          msg: msg.wasm.execute.msg,
+        },
+      })
+    } else if ('migrate' in msg.wasm) {
+      encoded = encodeRawProtobufMsg({
+        typeUrl: '/cosmwasm.wasm.v1.MsgMigrateContract',
+        value: {
+          sender: address,
+          contract: msg.wasm.migrate.contract_addr,
+          codeId: msg.wasm.migrate.new_code_id,
+          msg: msg.wasm.migrate.msg,
+        },
+      })
+    }
+  } else if ('stargate' in msg) {
+    encoded = {
+      typeUrl: msg.stargate.type_url,
+      value: fromBase64(msg.stargate.value),
+    }
+  } else {
+    console.error('Msg type not supported')
```

Comment:
Not sure what we want to do here if it's an unsupported message type. Think I covered all the options, but maybe we should throw an error?

Reply:
this fn does the same thing as `cwMsgToEncodeObject`, which we use to simulate
proposal messages :)